Automattic/simplenote-android
# Move scroll setup from view tree updates after the note content refreshes

Fixes #1338

The initial scroll position is set based on the last scroll position the user was or on the first keyword found if entering from search results. To set the scroll position, the app used a `ViewTreeObserver` with and set up a `OnGlobalFocusChangeListener` callback listener. The problem with this approach is that this callback gets called several times even after the text has been rendered. Thus, the scroll position for a search was overridden by the scroll position saved in preferences.


### Fix
We now change the scroll setup to after the note contents are refreshed. This happens when the note is loaded and set in the EditText. We post a task on the root view to be sure that the scroll setup is done after the text has been rendered. Also, when there is a configuration change, the note is not refreshed, but the view is re-rendered and the scroll position is lost. We setup the scroll as well when there is a configuration change.

### Test
**Note:** have a long note where the same term appears in different parts of the note

1. Have a note that contains some unique keyword (only once, only in one note)
2. Start search against this keyword
3. Tap the note ~> you might firstly see the keyword shown on screen
4. Go back to note list and tap the note again.
5. Tap to the next keyword
6. Rotate phone

Video from issue:

https://user-images.githubusercontent.com/73365754/110334627-a67b9c80-802b-11eb-9ee6-c25d524e220f.mp4

Fixed:

![20210824_005556](https://user-images.githubusercontent.com/195721/130563933-f8ee7ec6-7839-43ff-a471-c63064bb929f.gif)


### Review

Only one developer is required to review these changes, but anyone can perform the review.

### Release

`RELEASE-NOTES.txt` was updated in af046a10e74c078bc37f60700d2e2ece5c4e8fff with:
> Fixed scroll position of a note with matching keyword in not persistent
